semgrep-core/src/engine/Match_rules.ml

```diff
@@ -111,15 +112,30 @@ type env = {
   pattern_matches : id_to_match_results;
   (* used by metavariable-pattern to recursively call evaluate_formula *)
   file : Common.filename;
-  rule_id : string;
+  rule_id : R.rule_id;
   xlang : R.xlang;
   equivalences : Equivalence.equivalences;
+  (* problems found during evaluation, one day these may be caught earlier by
+   * the meta-checker *)
+  errors : E.error list ref;
 }
 
 (*****************************************************************************)
 (* Helpers *)
 (*****************************************************************************)
 
+(* Report errors during evaluation to the user rather than just logging them
+ * as we did before. *)
+let error env msg =
+  (* We are not supposed to report errors in the config file for several reasons
+   * (one being that it's often a temporary file anyways), so we report them on
+   * the target file. *)
+  let loc = PI.first_loc_of_file env.file in
+  let s = Printf.sprintf "rule %s: %s" env.rule_id msg in
+  (* TODO: warning or error? MatchingError or ... ? *)
```

Comment:
A little confused---it sounds like these used to be silently logged. Will reporting them as errors now cause them to crash Semgrep? Or is this just a cosmetic question?

Reply:
Not it won't crash Semgrep, they will just be communicated to the user instead of "silently" logged. Now that I'm looking into the test failures, I'm wondering if maybe not all of them should be communicated though!

Reply:
When running Semgrep you will e.g. see:
```
found problems analyzing 1 files; run with --verbose for details or run with --strict to exit non-zero if any file cannot be analyzed cleanly
```
And using `--verbose` you'll get:
```
semgrep-core reported a matching error
  --> matching internal error: rule test: metavariable-pattern failed because $Y it not in scope, please check your rule
```
But any matches that were found and reported previously should continue to be found and reported now.